code/obj/machinery/lightswitch.dm

```diff
@@ -49,6 +50,29 @@
 	if(!ON_COOLDOWN(src, "mechcomp_toggle", 1 SECOND))
 		toggle(null)
 
+/obj/machinery/light_switch/proc/autoposition()
+	var/turf/T = null
+	SPAWN(1 DECI SECOND)
```

Comment:
It might be slightly neater code to keep this spawn in `/obj/machinery/light_switch/auto/New()`, built switches don't need to delay for walls spawning in.